extensions/INTEL/SPV_INTEL_subgroup_buffer_prefetch.asciidoc

```diff
@@ -0,0 +1,174 @@
+= SPV_INTEL_subgroup_buffer_prefetch
+
+== Name Strings
+
+SPV_INTEL_subgroup_buffer_prefetch
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/KhronosGroup/SPIRV-Registry
+
+== Contributors
+
+// spell-checker: disable
+- Ben Ashbaugh, Intel +
+- Greg Lueck, Intel +
+- Andrzej Ratajewski, Intel +
+- Grzegorz Wawiorko, Intel +
+// spell-checker: enable
+
+== Notice
+
+Copyright (c) 2024 Intel Corporation.  All rights reserved.
+
+== Status
+
+- Complete
+
+== Version
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | {docdate}
+| Revision           | 1
+|========================================
+
+== Dependencies
+
+This extension is written against the SPIR-V Specification, Version 1.6 Revision 3.
+
+This extension requires SPIR-V 1.0.
+
+This extension extends the *SPV_INTEL_subgroups* extension and interacts with the *SPV_INTEL_cache_controls* extension.
+
+== Overview
+
+This extension extends the *SPV_INTEL_subgroups* extension by adding support for prefetching data from buffers.
+The functionality added by this extension can improve the performance of some kernels by prefetching data into a cache, so future reads of the data are from a fast cache rather than slower memory.
+
+== Extension Name
+
+To use this extension within a SPIR-V module, the appropriate *OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_INTEL_subgroup_buffer_prefetch"
+----
+
+== Modifications to the SPIR-V Specification, Version 1.6
+
+=== Capabilities
+
+Modify Section 3.31, Capability, adding rows to the Capability table:
+
+--
+[options="header"]
+|====
+2+^| Capability ^| Implicitly Declares 
+| 6220 | *SubgroupBufferPrefetchINTEL*
+|
+|====
+--
+
+=== Instructions
+
+Modify Section 3.49.21, Group and Subgroup Instructions, adding to the end of the list of instructions:
+
+[cols="1,1,2*3",width="100%"]
+|=====
+3+|[[OpSubgroupBlockPrefetchINTEL]]*OpSubgroupBlockPrefetchINTEL* +
+ +
+Prefetches one or more bytes from _Ptr_ for each invocation in the subgroup as a block operation, where the number of bytes to prefetch per invocation is specified by _NumBytes_.
+The total number of bytes that is collectively prefetched is therefore _NumBytes_ times _SubgroupSize_.
+Prefetching does not affect the functionality of a module but may change its performance characteristics. +
+ +
+_Ptr_ must be a pointer to an _integer type_ scalar in the *CrossWorkgroup* _Storage Class_. +
```

Comment:
Is it worth making this compatible with untyped pointers?

```suggestion
_Ptr_ must be a pointer. If its type is *OpTypePointer* it must point to an _integer type_ scalar in the *CrossWorkgroup* _Storage Class_. +
```

Don't forget to mention the interaction in the dependencies if this is desirable.

Reply:
Done - thanks!  Can you please check that I documented the interaction as you expected?

Noting for completeness: we also added an optional "Memory Operands" operand to pass alignment information, since we won't be able to infer alignment from the pointer type with untyped pointers.